lib/pbench/server/api/resources/upload_api.py

```diff
@@ -298,6 +304,24 @@ def put(self, filename: str):
 
         try:
             dataset.advance(States.UPLOADED)
+            Metadata.create(dataset=dataset, key=Metadata.SAVED, value=False)
+            Metadata.create(dataset=dataset, key=Metadata.SEEN, value=False)
```

Comment:
I think the metadata mechanism is fine...I'm just confused by the nuances of "columns" vs. "rows" here.  I naively expected that `Metadata` was a column, like "access" (however, that's spelled), and that, therefore, the `Metadata` column in a new table row would be assigned a default value (the way that the "access" column is).  If that were true, then I was hoping that we could engineer the default for that column to be something handy, like not-saved+not-seen, because the server already knows about those metadata keys and we know what their values should be for a newly-created dataset.

Are you saying that the `Metadata` column that I'm thinking of is actually a reference to a _row_ in another table, and that row has columns like `SEEN` and `SAVED`?  If so, then something still has to create that row when the row in the dataset table is created, and perhaps _it_ could apply these values as defaults.

Reply:
Some "metadata" is inherent in the `Dataset`, like `owner` and `access`. There's also a separate table allowing arbitrary client-controlled metadata in key/value pairs, which includes `seen` and `saved` flags, along with an expiration date; those rows are linked to the primary dataset via a foreign key. To some extent the distinction is arbitrary; in this case, it would have made sense to add a column for the expiration since it'll be used internally, but I chose to draw the line and add all the additional data to the metadata table instead.

Nevertheless, all of these are currently assigned values when a dataset is uploaded via `PUT`. Again, I could have drawn the line at the internally-used metadata and left `saved` and `seen` undefined until the dashboard creates values, but I chose to treat these specially.